### PR TITLE
Ron/p2p

### PR DESCRIPTION
**Pull request by Ronserruya**
Should be reviewed with the payment service PR : 
https://github.com/kinecosystem/kin-devplatform-payment-service/pull/1

A big number of changes is from vscode reformatting the code, I will make sure to disable that in the future :sob: 

PSA: The RSA private key added is just for testing, so its ok.

### CODE REVIEW

File: scripts/src/internal/services.ts [hunk 1]
```diff
@@ -169,16 +169,21 @@ export async function paymentComplete(payment: CompletedPayment, logger: LoggerI
 	}
 
 	order.setStatus("completed");
+	if (order.type !== "earn" && order.isExternalOrder()) {
```
Comment (mrsegev) [inline scripts/src/internal/services.ts]: Hardcoded strings "earn" ,"completed".  Can we move them to a dedicated file?

File: scripts/src/metrics.ts [hunk 2]
```diff
@@ -17,18 +17,18 @@ export function userActivate(newUser: boolean) {
 }
 
 export function timeRequest(time: number, method: string, path: string) {
-	statsd.timing("request", time, undefined, { path: `${ method }: ${ path }` });
+	statsd.timing("request", time, undefined, { path: `${method}: ${path}` });
 }
 
-export function createOrder(orderType: "marketplace" | "external", offerType: "earn" | "spend", offerId: string) {
+export function createOrder(orderType: "marketplace" | "external", offerType: "earn" | "spend" | "pay_to_user", offerId: string) {
```
Comment (mrsegev) [inline scripts/src/metrics.ts]: Again, hardcoded strings. Let's talk about them.

File: scripts/src/internal/services.ts [hunk 3]
```diff
@@ -169,16 +169,21 @@ export async function paymentComplete(payment: CompletedPayment, logger: LoggerI
 	}
```
Comment (yosriz) [inline scripts/src/internal/services.ts]: at line 160 there are comments regards p2p I guess there are already done by this PR, so please delete if so

File: scripts/src/models/orders.ts [hunk 4]
```diff
@@ -188,12 +189,18 @@ export class Order extends CreationDateModel {
...
+	public recipientId?: string;
+
 	@Column({ name: "offer_id" })
 	public offerId!: string;
 
 	@Column("simple-json")
 	public meta!: OrderMeta;
 
+	@Column("simple-json", { nullable: true })
+	public recipientMeta?: OrderMeta;
```
Comment (yosriz) [inline scripts/src/models/orders.ts]: rename to "recipient_meta" to follow db column names convention

File: scripts/src/public/services/orders.ts [hunk 5]
```diff
@@ -158,23 +160,42 @@ export async function createExternalOrder(jwt: string, user: User, logger: Logge
...
+		let recipient_title: string | undefined;
+		let recipient_description: string | undefined;
 		if (payload.sub === "earn") {
 			title = (payload as ExternalEarnOrderJWT).recipient.title;
 			description = (payload as ExternalEarnOrderJWT).recipient.description;
 			sender_address = app.walletAddresses.sender;
 			recipient_address = user.walletAddress;
-		} else {
+		} else if (payload.sub === "spend") {
 			// spend or pay_to_user
```
Comment (yosriz) [inline scripts/src/public/services/orders.ts]: delete this comment

File: scripts/src/public/services/orders.ts [hunk 6]
```diff
@@ -158,23 +160,42 @@ export async function createExternalOrder(jwt: string, user: User, logger: Logge
...
 			await addWatcherEndpoint([app.walletAddresses.recipient]);  // XXX how can we avoid this and only do this for the first ever time we see this address?
 			title = (payload as ExternalSpendOrderJWT).sender.title;
 			description = (payload as ExternalSpendOrderJWT).sender.description;
 			sender_address = user.walletAddress;
 			// TODO in case of pay_to_user, needs another lookup for the recipient_user_wallet
```
Comment (yosriz) [inline scripts/src/public/services/orders.ts]: comment not relevant

**Pull request merged**